src/main/java/org/folio/mr/service/impl/EcsRequestServiceImpl.java

```diff
@@ -1,50 +1,103 @@
+import org.folio.mr.service.CloningService;
 import org.folio.mr.service.ConsortiumService;
 import org.folio.mr.service.EcsRequestService;
+import org.folio.mr.service.UserService;
 import org.folio.spring.service.SystemUserScopedExecutionService;
 import org.springframework.stereotype.Service;
 
-import lombok.RequiredArgsConstructor;
-import lombok.extern.log4j.Log4j2;
+import java.util.UUID;
 
 @Service
 @Log4j2
 @RequiredArgsConstructor
 public class EcsRequestServiceImpl implements EcsRequestService {
 
+  private final FakePatronLinkRepository fakePatronLinkRepository;
   private final EcsTlrClient ecsTlrClient;
+  private final UserService userService;
+  private final CloningService<User> userCloningService;
   private final SystemUserScopedExecutionService executionService;
   private final ConsortiumService consortiumService;
 
   @Override
   public EcsTlr create(MediatedRequestEntity mediatedRequest) {
-    log.info("confirm:: creating ECS TLR for mediated request {}", mediatedRequest::getId);
-    return executionService.executeSystemUserScoped(consortiumService.getCentralTenantId(),
-      () -> ecsTlrClient.post(buildEcsTlr(consortiumService.getCurrentTenantId(),
-        mediatedRequest)));
+    log.info("create:: Creating fake patron for mediated request {}", mediatedRequest.getId());
+    String fakeUserId = createFakePatron(mediatedRequest);
+
+    log.info("create:: Creating fake patron link between real patron {} and fake patron {}",
+      mediatedRequest.getRequesterId(), fakeUserId);
+    createFakePatronLink(mediatedRequest.getRequesterId(), fakeUserId);
+
+    log.info("create:: Creating ECS request for fake patron {}, mediated request {}", fakeUserId,
+      mediatedRequest.getId());
+    return createEcsTlr(mediatedRequest, fakeUserId);
+  }
+
+  private String createFakePatron(MediatedRequestEntity mediatedRequest) {
+    User realPatron = userService.fetchUser(asString(mediatedRequest.getRequesterId()));
+    User fakePatron = buildFakePatron(realPatron.getPatronGroup());
+
+    log.info("createFakePatron:: Creating local fake patron");
+    User localFake = userService.create(fakePatron);
+
+    log.info("createFakePatron:: Creating remote fake shadow patron: ID {}", localFake.getId());
+    executionService.executeSystemUserScoped(consortiumService.getCentralTenantId(),
+      () -> userCloningService.clone(localFake));
+
+    return localFake.getId();
   }
 
-  private static EcsTlr buildEcsTlr(String primaryRequestTenantId,
-    MediatedRequestEntity mediatedRequest) {
+  private User buildFakePatron(String patronGroupId) {
+    return new User()
+      .active(true)
+      .barcode(format("securepatron_%s", UUID.randomUUID()))
+      .type("patron")
+      .patronGroup(patronGroupId)
+      .personal(
+        new UserPersonal()
+          .preferredContactTypeId("002")
+          .firstName("Secure")
+          .lastName("Patron")
+          .email(UUID.randomUUID() + "@example.com")
```

Comment:
Could be the same as barcode, `@example.com` part is not necessary

Reply:
done